### DISCUSSION TITLE
v1.20.0

### DISCUSSION THREAD

**Discussion by romange** [dragonflydb/dragonfly#3294] [Announcements]
# Dragonfly v1.20.0

Some prominent changes include:
* Improvements around client side caching (#3136 #3158)
  Specifically, rueidis is now supported (#2454)
* Improvements around search and vector search APIs, FT.ALTER is now supported (#3144, #3148, #3178, #3186)
* JSON.MSET is added (#3167)
* Expiry notifications are now supported (#3154)

## What's Changed
* test(cluster_mgr): Add tests for cluster_mgr.py by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3129
* chore: get rid of kv_args and replace it with slices to full_args by @romange in https://github.com/dragonflydb/dragonfly/pull/3101
* fix: Fix live-lock in connection test by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3135
* chore: call breaker_cb_ on shutdown by @romange in https://github.com/dragonflydb/dragonfly/pull/3128
* chore: Add option to disable aws dependency by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3077
* feat(cluster): Add important logs by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3138
* chore: a small cleanup by @romange in https://github.com/dragonflydb/dragonfly/pull/3140
* test: skip test_cluster_migration_cancel, it is broken by @BorysTheDev in https://github.com/dragonflydb/dragonfly/pull/3146
* chore: update helio by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3150
* fix: fix bug in cluster/slot_set by @BorysTheDev in https://github.com/dragonflydb/dragonfly/pull/3143
* chore: recommit client tracking by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3136
* feat(search): Tag field options (separator, case sensitivity) by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3144
* feat(search): basic FT.ALTER by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3148
* chore: Introduce pipeline back-pressure by @romange in https://github.com/dragonflydb/dragonfly/pull/3152
* fix: lua and client tracking by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3163
* feat: client tracking optout by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3158
* feat: add noloop subcommand in client tracking by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3164
* fix(ci): docker not using iouring by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3169
* fix(ci): add missing docker option on reg tests workflow by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3172
* fix: acl compatibility by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3147
* chore: Streamer is rewritten with async interface by @romange in https://github.com/dragonflydb/dragonfly/pull/3108
* feat(json): MSET by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3167
* fix(cluster): Wait for flow creation before cancelling it by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3159
* chore: improve ft.create for vector search by @romange in https://github.com/dragonflydb/dragonfly/pull/3178
* feat: add incoming migration error processing by @BorysTheDev in https://github.com/dragonflydb/dragonfly/pull/3175
* chore: add parsing support for the rest of VECTOR parameters by @romange in https://github.com/dragonflydb/dragonfly/pull/3179
* fix(cluster): Wait before all access to slot migrations by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3180
* fix(server): fix bug in replication on cached mode by @adiholden in https://github.com/dragonflydb/dragonfly/pull/3156
* chore: small acl compat changes by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3177
* chore: allow calling Context::ReportError from I/O dispatch fiber by @romange in https://github.com/dragonflydb/dragonfly/pull/3182
* chore: add a lexer test covering KNN query by @romange in https://github.com/dragonflydb/dragonfly/pull/3185
* chore(search): Add __vector_score alias by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3186
* chore: Refactor string span management by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3165
* chore(ci): run replication tests on arm by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3168
* chore(search): improve parser error reporting by @romange in https://github.com/dragonflydb/dragonfly/pull/3184
* fix(reply_builder): remove virtual modifier in SendError(ErrorReply) method by @BagritsevichStepan in https://github.com/dragonflydb/dragonfly/pull/3191
* fix(cluster): Support `FLUSHALL` while slot migration is in progress by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3173
* fix: fix RegisterOnChange methods for journal and db_slice by @BorysTheDev in https://github.com/dragonflydb/dragonfly/pull/3171
* fix(generic_family): fix RenameGeneric command for non-string data types by @BagritsevichStepan in https://github.com/dragonflydb/dragonfly/pull/3181
* fix(server): Rename confusing flag `replica_reconnect_on_master_restart` by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3193
* fix(unit tests): fix generic family info test by @adiholden in https://github.com/dragonflydb/dragonfly/pull/3187
* test: improve cluster_fuzzy_migration test by @BorysTheDev in https://github.com/dragonflydb/dragonfly/pull/3197
* chore(core): Remove DfImpl inside ScoredMap by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3199
* chore(tiering): Remove IoMgr by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3198
* test(cluster): Make sure migration maintains TTL by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3188
* feat(cluster): Support `STICK` bit in slot migration by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3200
* chore: Re-enable previously flaky test by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3196
* feat(acl): add support of multiple passwords by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3189
* chore(tiering): More advanced tiering tests by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3201
* chore: add replica-priority flag by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3204
* chore: fix tiering regtest test by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3208
* feat(generic_family): Assemble RESTORE and STICK commands into one commmand during replication of the RENAME command by @BagritsevichStepan in https://github.com/dragonflydb/dragonfly/pull/3209
* core(search): Add EF_RUNTIME parameter by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3206
* fix(sanitizers): failing json test by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3176
* feat(server): expiry notifications by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3154
* fix: replicaof_reject_on_load flake by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3203
* fix(cluster): Don't miss keys when migrating slots by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3218
* fix(tiering): Fix pending leak during immediate stash error by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3219
* chore: provide basic logging to catch possible command errors by @romange in https://github.com/dragonflydb/dragonfly/pull/3213
* chore: disable tiering test by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3236
* fix(transaction): Don't transactionalize empty EVAL inside EXEC by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3231
* fix: fix deadlock in DFLYCLUSTER CONFIG command and outgoing migration finaliztion by @BorysTheDev in https://github.com/dragonflydb/dragonfly/pull/3239
* chore(tiered): minor fixes + expose buffer allocation type stats by @romange in https://github.com/dragonflydb/dragonfly/pull/3234
* fix(pytest): timed ticker for simpler conditions by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3242
* fix(server): Fix SCAN deadlock by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3235
* chore: introduce back-pressure to tiered storage by @romange in https://github.com/dragonflydb/dragonfly/pull/3243
* chore: improve dfly_bench by @romange in https://github.com/dragonflydb/dragonfly/pull/3244
* chore(tiering): External alloc free page chaining by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3237
* fix: total_stash_overflows statistic by @romange in https://github.com/dragonflydb/dragonfly/pull/3248
* chore: update action versions by @romange in https://github.com/dragonflydb/dragonfly/pull/3250
* chore(lua): Return which undeclared key was accessed by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3245
* chore: replace session wide fixtures with scope by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3251
* chore: pull helio and adjust mimalloc for 32MiB segments by @kostasrim in https://github.com/dragonflydb/dragonfly/pull/3174
* chore: add more logs around the duplicate value case in small_bins by @romange in https://github.com/dragonflydb/dragonfly/pull/3254
* fix: fix move error during migration finalization by @BorysTheDev in https://github.com/dragonflydb/dragonfly/pull/3253
* chore: upload all the logs from /tmp by @romange in https://github.com/dragonflydb/dragonfly/pull/3263
* fix(tiering): Throttle snapshot load by @dranikpg in https://github.com/dragonflydb/dragonfly/pull/3249
* chore: initiate grow preemptively by @romange in https://github.com/dragonflydb/dragonfly/pull/3267
* chore(monitoring): add more dashboards + memcached by @romange in https://github.com/dragonflydb/dragonfly/pull/3268
* fix(json_family): fix JSON.GET commmand for JSON legacy mode by @BagritsevichStepan in https://github.com/dragonflydb/dragonfly/pull/3261
* chore: dfly_bench can send traffic to memcached by @romange in https://github.com/dragonflydb/dragonfly/pull/3271
* chore: print effective QPS of the server. by @romange in https://github.com/dragonflydb/dragonfly/pull/3274
* chore: debugging fixes by @romange in https://github.com/dragonflydb/dragonfly/pull/3275
* chore: more debug checks around tiered storage by @romange in https://github.com/dragonflydb/dragonfly/pull/3277
* chore(license): Fix typo in LICENSE.md text by @softmoth in https://github.com/dragonflydb/dragonfly/pull/3283
* fix(test): Verify that save has not finished by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3278
* Create scorecard.yml by @romange in https://github.com/dragonflydb/dragonfly/pull/3270
* chore: Disable cluster_mgr_test by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3279
* fix: Increase key count to make test more robust by @chakaz in https://github.com/dragonflydb/dragonfly/pull/3286
* fix: properly clean tiered state upon flash by @romange in https://github.com/dragonflydb/dragonfly/pull/3281
* fix(bug): crash on takeover and info replication by @adiholden in https://github.com/dragonflydb/dragonfly/pull/3282

## New Contributors
* @softmoth made their first contribution in https://github.com/dragonflydb/dragonfly/pull/3283

## Huge thanks to all the contributors! ❤️

**Full Changelog**: https://github.com/dragonflydb/dragonfly/compare/v1.19.0...v1.20.0

<hr /><em>This discussion was created from the release <a href='https://github.com/dragonflydb/dragonfly/releases/tag/v1.20.0'>v1.20.0</a>.</em>

**Comment by kostasrim** [comment]
:rocket: :rocket: :rocket:

**Comment by ishergin** [comment]
Why AWS snapshots were disabled by default? https://github.com/dragonflydb/dragonfly/pull/3077

**Reply by romange** [reply]
1.20.1 was released and it fixes this issue
